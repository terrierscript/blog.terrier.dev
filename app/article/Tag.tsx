import React, { SFC } from "react"
import styled from "styled-components"
import { kebabCase } from "lodash"
import Link from "gatsby-link"

export const TagItem = styled.div`
  display: inline-block;
  background: #e08e79;
  color: #555;
  padding: 0.2em 0.5em 0.3em;
  margin: 0.2em;
  border-radius: 20px;
  font-size: 0.8em;
  text-transform: lowercase;
  ::before {
    content: "# ";
  }
`

export const Tag: SFC<{ tag: string }> = ({ tag }) => {
  return (
    <Link to={`/tags/${kebabCase(tag)}/`}>
      <TagItem>{tag}</TagItem>
    </Link>
  )
}
