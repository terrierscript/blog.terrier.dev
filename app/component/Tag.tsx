import React, { SFC } from "react"
import styled from "styled-components"
import { Link } from "gatsby"
import { headerFont } from "../utils/typography"

export const TagItem = styled.div`
  display: inline-block;
  background: #e08e79;
  color: #000;
  font-family: ${headerFont};
  font-weight: bold;
  padding: 0.2em 0.5em 0.3em;
  margin: 0.2em;
  border-radius: 20px;
  font-size: 0.8em;
  text-transform: lowercase;
  ::before {
    content: "# ";
  }
`

export const Tag: SFC<{ tag: string }> = ({ tag }) => {
  return (
    <Link to={`/tags/${tag.toLowerCase()}/`}>
      <TagItem>{tag}</TagItem>
    </Link>
  )
}
