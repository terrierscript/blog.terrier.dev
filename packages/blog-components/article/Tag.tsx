import React, { SFC } from "react";
import styled from "styled-components";
import { kebabCase } from "lodash";
import Link from "gatsby-link";

const TagItem = styled.div`
  display: inline-block;
  background: #ccc;
  color: #555;
  padding: 0.2em 0.5em;
  margin: 0.2em;
  border-radius: 20px;
  font-size: 0.8em;
`;

export const Tag: SFC<{ tag: string }> = ({ tag }) => {
  return (
    <Link to={`/tags/${kebabCase(tag)}/`}>
      <TagItem>{tag}</TagItem>
    </Link>
  );
};
