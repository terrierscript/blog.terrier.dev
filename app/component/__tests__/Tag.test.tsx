import React from "react"
import renderer from "react-test-renderer"
import { AutoLinkTag } from "../Tag"

test("<Tag>", () => {
  const tree = renderer.create(<AutoLinkTag tag={"foo"} />).toJSON()
  expect(tree).toMatchSnapshot()
})
