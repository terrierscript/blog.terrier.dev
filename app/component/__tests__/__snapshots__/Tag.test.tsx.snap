// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag> 1`] = `
<a
  href="/tags/foo/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="css-emcna"
  >
    foo
  </div>
</a>
`;
